src/gflownet/utils/metrics.py

```diff
@@ -158,6 +149,14 @@ def sharpeRatio(p, Q, x, rf):
 
 def _sharpeRatioQPMax(p, Q, rf):
     """ Sharpe ratio maximization problem - QP formulation """
+    from cvxopt import matrix
```

Comment:
Right, this shouldn't have ended up here, I moved the import inside the function since on my local environment (with M1 chip) cvxopt was really annoying to install and I ended up simply moving forward without it is only used in that one function.

I can revert it but at the same time since having the import down there removes a hard requirement which could be problematic for several users it could make sense to keep it there. What do you think?

Reply:
Makes sense, I'd recommend adding a comment explaining this and opening an issue on github so we're reminded to check back when cvxopt fixes their M1 compatibility (and so that other potential M1 users facing this problem might see it).

Reply:
See [this issue](https://github.com/recursionpharma/gflownet/issues/56).